lib/rules/named-spy.js

```diff
@@ -9,14 +9,33 @@ function findIdentifier (node) {
   var parent = node.parent
   while (parent) {
     if (parent.type === 'VariableDeclarator') {
-      return parent.id
+      return {
+        node: parent.id,
+        name: parent.id.name
+      }
     } else if (parent.type === 'Property') {
-      return parent.key
+      if (parent.key.type === 'Identifier') {
```

Comment:
I don't think these nested conditionals are a good pattern (hard to read and mantain), could you refactor to a one level conditional structure or extract functions so that the conditions are clearer?

Reply:
Agree. I'll make it one level and add function for comparison.